### ISSUE TITLE
Incorrect NS4000 when using typeof expression

### ISSUE THREAD

**Issue by samcragg** [nsubstitute/NSubstitute.Analyzers#71]
I think the analyzer is incorrectly firing `NS4000: Returns() is set with a method that itself calls Returns` for the following code (this is the simplest I could get the code to reproduce the warning I was getting):

```C#
public interface IExample
{
    object Method();
}

public class ExampleTest
{
    [Fact]
    public void TypeofTestClass()
    {
        IExample example = Substitute.For<IExample>();
        example.Method().Returns(typeof(ExampleTest));
        //                       ^^^^^^^^^^^^^^^^^^^
    }
}
```

This is under dotnet-core 2.1 with using NSubstitute version 4.0.0, NSubstitute.Analyzers.CSharp version 1.0.5 and xunit version 2.4.1

Granted, the above code doesn't look realistic (there's no asserts!), however, I was getting the warning in one of my actual tests and it seems to be the `typeof(TheClassContainingTheTest)` that triggers it.

Let me know if there's anything I can do to help.

**Comment by tpodolak** [comment]
@samcragg thanks for reporting, this one for sure is a bug. Fix is waiting for review

**Comment by tpodolak** [comment]
@samcragg fix will be available in next release

**Comment by tpodolak** [comment]
@samcragg released in [1.0.6](https://www.nuget.org/packages/NSubstitute.Analyzers.CSharp/1.0.6)

**Comment by samcragg** [comment]
Just to confirm that's fixed it, thanks - that was an amazingly quick turnaround!
